app/models/company.rb

```diff
@@ -0,0 +1,18 @@
+class Company < ActiveRecord::Base
+  has_many :items
+
+  def self.company_save
+    all_companies = FHServices.new.companies_hash[:companies]
+    all_companies.each do |company|
+      unless Company.find_by(shortname: company[:shortname])
+        Company.create(name: company[:name], shortname: company[:shortname])
+      end
+    end
+    #how do I account for deleted companies?
```

Comment:
Good question. My first instinct is to add a new field to the Company model. Perhaps a boolean `record_removed?` which is scoped out by `Company.in_record`. You could mark the companies in this method by comparing a cached collection of companies with the all_companies.

This way you retain the record in case you need it for anything else.